### PR TITLE
Feature/dynamic loading

### PR DESCRIPTION
**Pull request by tgriff3433**
Came across a fix for local development (yarn dev), causing issues with web3provider during hot reload.

### CODE REVIEW

File: src/components/defaultprovider.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { createWeb3ReactRoot } from '@web3-react/core'
```
Comment (wpmonty) [inline src/components/defaultprovider.js:1]: rename file to `Web3ReactProviderDefaultSSR.ts` and move to `src/providers`

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -11,7 +11,10 @@ function getLibrary(provider) {
   return new ethers.providers.Web3Provider(provider) // this will vary according to whether you use e.g. ethers or web3.js
 }
 
-const Web3ReactProviderDefault = createWeb3ReactRoot(DefaultProviderName)
+const Web3ReactProviderDefault = dynamic(
+  () => import('../components/defaultprovider'),
+  { ssr: false }
```
Comment (wpmonty) [inline src/pages/_app.tsx:16]: this essentially kills SSG & SSR for the entire site, correct?

The way we're headed with CMS discussion would fundamentally require SSG, so we may need to pull on the thread of serving web3-dehydrated from server site and hydrating web3 thereafter. 

Maybe we return `<SiteLayout web3={false}>` from the server as default and use the prop downstream to disable any web3 components like the nav button or claim page. cc @nateawelch

Comment (eenagy) [reply src/pages/_app.tsx:16]: Both SSG & SSR will still be supported with dynamic.

Comment (tgriff3433) [reply src/pages/_app.tsx:16]: Im having a difficult time looking this up actually, ssr: false does disable ssr but I don't know if the nested components inherit this. do you have a way of proving ssr still works through the rest of the app when the parent component is dynamically loaded?

Comment (tgriff3433) [reply src/pages/_app.tsx:16]: I can confirm that .next/server/pages does have the pre rendered html files. but I don't know enough about react to say if that is conclusive.

**Review by wpmonty — APPROVED** [review]

**Comment by wpmonty** [comment]
SSR seems unaffected; merging!

**Pull request merged**